README.md

```diff
@@ -2,92 +2,133 @@
 
+All of the sub-protocols in the Portal Network establish their own Kademlia DHT network that is managed independent of the base Discovery V5 protocol Kademlia DHT.
 
-### JSON-RPC API
+
+## Design Principles
+
+Each of the Portal Network sub-protocols follows these design principles.
+
+1. Isolation
+  - Each network should be able to be run independent of the other networks.  Participation in one network should not require participation in another network.
+2. Distribution of Responsibility
+  - Normal operation of the network should result in a roughly even spread of responsibility across the individual nodes in the network.
+3. Tunable Resource Requirements
+  - Individual nodes should be able to control the amount of machine resources (HDD/CPU/RAM) they must expend to be part of the network.
+
+These design principles are aimed at ensuring that participation in the Portal Network is feasible even on resource constrained devices.
+
+## The JSON-RPC API
 
 The following JSON-RPC API endpoints are intended to be supported by the portal network and exposed by portal clients.
 
 TODO
 
+## Bridge Nodes
+
+We use the term "bridge" node to refer to the process by which data enters the Portal Network.  A "bridge" node is a normal node in the Portal Network that is able to bring in new and missing data from some external source and inject it into the Portal Network.  These nodes would not be special in any way from the perspective of the Portal Network.
+
+The planned architecture for bridge nodes is to pull data from the standard JSON-RPC API of a Full Node and "push" this data into their respective networks within the Portal Network.
+
 ## Network Functionality
 
-### State: Accounts and Contract Storage
+### State Network: Accounts and Contract Storage
+
+The State Network facilitates on-demand retrieval of the Ethereum "state" data.  This includes things like:
 
-The state network facilitates on-demand retrieval of the Ethereum "state" data.
+- Reading account balances or nonce values
+- Retrieving contract code
+- Reading contract storage values
 
-Nodes should be able to choose how much state they want to store and share, and the network should provide a way to identify which nodes to query for a wanted portion of state. This is so that every node, no matter how small, can contribute to the health and robustness of the network.
+The responsibility for storing the underlying "state" data should be evenly distributed across the nodes in the network.  Nodes must be able to choose how much state they want to store.  The data is distributed in a manner that allows nodes to determine the appropriate nodes to query for any individual piece of state data.  When retrieving state data, a node should be able to validate the response using a recent header from the header chain.
```

Comment:
I'm still not sure how to phrase it but "nodes can choose how much to store" and "the responsibility to storing state is split evenly" can only be unified by finding the node which has chosen the smallest amount of data and giving that amount of data to every node.

I don't think anything is lost by dropping "The responsibility for storing the underlying "state" data should be evenly distributed across the nodes in the network.", but I could be misunderstanding!

Also, for "When retrieving state data, a node should be able to validate the response using a recent header from the header chain.", the responding node won't provide any proofs?

Reply:
Maybe the problem here is distinction between individual nodes and the network as a whole.

- we want the data distributed evenly round the network...
- we accept that the data is not evenly distributed across individual nodes...
- we aim for an emergent property which is that nodes that offer the same storage will have roughly the same amount of the state data.

Still going to leave this for a later editorial pass.

Reply:
> Also, for "When retrieving state data, a node should be able to validate the response using a recent header from the header chain.", the responding node won't provide any proofs?

Yeah, wording isn't totally crystal clear here.  Yes, some responses come back with a proof, and they would validate that proof by checking that it is anchored to a recent state root from the header chain...